src/BenchmarksServer/Startup.cs

```diff
@@ -1391,54 +1415,107 @@ private static Process StartProcess(string hostname, string benchmarksRepo, Serv
                     Log.WriteLine(e.Data);
                     standardOutput.AppendLine(e.Data);
 
-                    if (job.State == ServerState.Starting && (e.Data.ToLowerInvariant().Contains("started") || e.Data.ToLowerInvariant().Contains("listening")))
+                    if (job.State == ServerState.Starting &&
+                        (e.Data.ToLowerInvariant().Contains("started") || e.Data.ToLowerInvariant().Contains("listening")))
                     {
-                        job.StartupMainMethod = stopwatch.Elapsed;
+                        MarkAsRunning(hostname, benchmarksRepo, job, perfview, stopwatch, process);
+                    }
+                }
+            };
 
-                        Log.WriteLine($"Running job '{job.Id}' with scenario '{job.Scenario}'");
-                        job.Url = ComputeServerUrl(hostname, job);
+            stopwatch.Start();
+            process.Start();
+            process.BeginOutputReadLine();
 
-                        // Start perfview?
-                        if (perfview)
-                        {
-                            job.PerfViewTraceFile = Path.Combine(job.BasePath, "benchmarks.etl");
-                            var perfViewArguments = new Dictionary<string, string>();
-                            perfViewArguments["AcceptEula"] = "";
-                            perfViewArguments["NoGui"] = "";
-                            perfViewArguments["Process"] = process.Id.ToString();
+            if (iis)
+            {
+                // IIS doesn't print anything
+                Thread.Sleep(1000);
```

Comment:
I would like to figure out a better way to detect if IIS is up or not. This is before first request, correct? Were you running into issues sending a request at this point or were there issues with perfView?

Reply:
Yea, we should send a request. This is similar to what docker does. See https://github.com/aspnet/benchmarks/blob/557be1ebf14297ff0910f905ddd38bb30c99b514/src/BenchmarksServer/Startup.cs#L684-L700

Reply:
We shouldn't send requests or we would skew first request time statistics. Socket being able to connect is enough.